### PR TITLE
Edit mentor profile

### LINKED ISSUE #258: API: Update mentor profile
Create an endpoint POST `/api/profile/mentor` that will update the following information from the body:
- name
- timeZone
- skills: array of numbers, ids of skills.

When updating skills, simply delete all existing skills in `mentor_skills` for that mentor and recreate all of them from the body.

To get the user's id, pass `req.user` from router to controller and use `req.user.id`.

Bonus points:
- validate that user exists and return 404 if it doesn't
- validate that name is not an empty string and return 400 if it is
- validate that the user is a student and return 404 if it isn't.

See instructions on the authentication in #255.

### PR DESCRIPTION
**Pull request by AnilMula**
# Description

The PR is to update mentor profile. It updates name, time zone and skills.

Fixes #258 

# How to test?

It can be tested with postman by providing authentication header

# Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] This PR is ready to be merged and not breaking any other functionality

### CODE REVIEW

File: src/server/api/controllers/profile.controller.js [hunk 1]
```diff
@@ -37,7 +38,63 @@ const getAdminsProfile = async (userId) => {
...
   }
 };
+
+const editMentorProfile = async (mentorId, updatedMentor) => {
+  // validate that user exists and return 404 if it doesn't
+  const getUser = await knex('users').select('id').where('id', mentorId);
+  if (getUser[0].id !== mentorId) return 404;
+
+  // validate that name is not an empty string and return 400 if it is
+  if (updatedMentor.name === null) return 400;
```
Comment (tpetrina) [inline src/server/api/controllers/profile.controller.js]: You could write:
```suggestion
  if (!updatedMentor.name) return 400;
```
This covers `null`, `undefined` and `''`

**Review by tpetrina — DISMISSED** [review]
Looks great, have you tested it successfully?

**Comment by AnilMula** [comment]
Yes I tested with postman

**Review by tpetrina — APPROVED** [review]
Good job

**Pull request merged**